pkg/source/resource_test.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2022 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package source_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	diecorev1 "dies.dev/apis/core/v1"
+	diemetav1 "dies.dev/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest/fake"
+	"k8s.io/client-go/restmapper"
+	k8sscheme "k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/google/go-cmp/cmp"
+	cartov1alpha1 "github.com/vmware-tanzu/apps-cli-plugin/pkg/apis/cartographer/v1alpha1"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime"
+	clitesting "github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/testing"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/source"
+)
+
+func TestWorkloadOptionsFetchResourceObjects(t *testing.T) {
+	defaultNamespace := "default"
+	workloadName := "my-workload"
+	pod1Die := diecorev1.PodBlank.
+		MetadataDie(func(d *diemetav1.ObjectMetaDie) {
+			d.Name("pod1")
+			d.Namespace(defaultNamespace)
+			d.AddLabel(cartov1alpha1.WorkloadLabelName, workloadName)
+		}).Kind("pod")
+	givenWorkload := &cartov1alpha1.Workload{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: defaultNamespace,
+			Name:      workloadName,
+		},
+	}
+	scheme := runtime.NewScheme()
+	c := cli.NewDefaultConfig("test", scheme)
+	c.Client = clitesting.NewFakeCliClient(clitesting.NewFakeClient(scheme))
```

Comment:
keep the client and delete the new cli config.

```suggestion
fakeClient := clitesting.NewFakeCliClient(clitesting.NewFakeClient(scheme))
```